docs/docs/reference/changed/structural-types.md

```diff
@@ -3,131 +3,59 @@ layout: doc-page
 title: "Programmatic Structural Types"
 ---
 
-Previously, Scala supported structural types by means of
-reflection. This is problematic on other platforms, because Scala's
-reflection is JVM-based. Consequently, Scala.js and Scala.native don't
-support structural types fully. The reflection based implementation is
-also needlessly restrictive, since it rules out other implementation
-schemes. This makes structural types unsuitable for e.g. modelling
-rows in a database, for which they would otherwise seem to be an ideal
-match.
+Some usecases, such as modelling database access, are more awkward in
+statically typed languages than in dynamically typed languages: With
+dynamically typed languages, it's quite natural to model a row as a
+record or object, and to select entries with simple dot notation (e.g.
+`row.columnName`).
 
-Dotty allows to implement structural types programmatically, using
-"Selectables". `Selectable` is a trait defined as follows:
+Achieving the same experience in statically typed
+language requires defining a class for every possible row arising from
+database manipulation (including rows arising from joins and
+projections) and setting up a scheme to map between a row and the
+class representing it.
 
-    trait Selectable extends Any {
-      def selectDynamic(name: String): Any
-      def selectDynamicMethod(name: String, paramClasses: ClassTag[_]*): Any =
-        new UnsupportedOperationException("selectDynamicMethod")
-    }
+This requires a large amount of boilerplate, which leads developers to
+trade the advantages of static typing for simpler schemes where colum
+names are represented as strings and passed to other operators (e.g.
+`row.select("columnName")`). This approach forgoes the advantages of
+static typing, and is still not as natural as the dynamically typed
+version.
 
-The most important method of a `Selectable` is `selectDynamic`: It
-takes a field name and returns the value associated with that name in
-the selectable.
+Structural types help in situations where we would like to support
+simple dot notation in dynamic contexts without losing the advantages
+of static typing. They allow developers to use dot notation and
+configure how fields and methods should be resolved.
 
-Assume now `r` is a value with structural type `S`. In general `S` is
-of the form `C { Rs }`, i.e. it consists of a class reference `C` and
-refinement declarations `Rs`. We call a field selection `r.f`
-_structural_ if `f` is a name defined by a declaration in `Rs` whereas
-`C` defines no member of name `f`. Assuming the selection has type
-`T`, it is mapped to something equivalent to the following code:
+## Example
 
-    (r: Selectable).selectDynamic("f").asInstanceOf[T]
+<script src="https://scastie.scala-lang.org/Duhemm/HOZFKyKLTs294XOSYPU5Fw.js"></script>
```

Comment:
I would not link to `Scastie` since it only supports a very outdated version of Dotty. Just inline the example in the documentation